common/reactive/src/main/java/io/helidon/common/reactive/Multi.java

```diff
@@ -178,6 +181,44 @@
         return new MultiFromStream<>(stream);
     }
 
+    /**
+     * Create a {@link Multi} instance that publishes {@link ByteBuffer}s from the given {@link InputStream}.
+     * <p>
+     * If {@link InputStream} can be trusted not to block on read operations, use faster
+     * {@link Multi#from(java.io.InputStream, int)}.
+     * <p>
+     * The operator calls {@link InputStream#close()} when the cancel signal is intercepted.
+     *
+     * @param inputStream the Stream to publish
+     * @param buffSize size of the chunks to be sent to downstream in the form of {@link ByteBuffer}
+     * @param executor executor to use for waiting at {@link InputStream}'s blocking reads
+     * @return Multi
+     * @throws NullPointerException if {@code stream} is {@code null}
+     */
+    static Multi<ByteBuffer> from(InputStream inputStream, int buffSize, ExecutorService executor) {
+        Objects.requireNonNull(inputStream, "stream is null");
+        Objects.requireNonNull(executor, "executor is null");
+        return new MultiFromBlockingInputStream(inputStream, buffSize, executor);
+    }
+
+    /**
+     * Create a {@link Multi} instance that publishes {@link ByteBuffer}s from the given {@link InputStream}.
+     * <p>
+     * {@link InputStream} is trusted not to block on read operations, in case it can't be assured use
+     * {@link Multi#from(java.io.InputStream, int, java.util.concurrent.ExecutorService)}.
+     * <p>
+     * The operator calls {@link InputStream#close()} when the cancel signal is intercepted.
+     *
+     * @param inputStream the Stream to publish
+     * @param buffSize    size of the chunks to be sent to downstream in the form of {@link ByteBuffer}
+     * @return Multi
+     * @throws NullPointerException if {@code stream} is {@code null}
+     */
+    static Multi<ByteBuffer> from(InputStream inputStream, int buffSize) {
```

Comment:
The `buffSize` in both methods seems a bit of an overhead.
I think the static method `from` should only accept `InputStream` (and `ExecutorService` for blocking variant).
If you want to have the buffSize, a builder may be the right way to go, using some reasonable default for buffer size otherwise.

Reply:
We can do something like this:
```java
        String s = Multi.from(is, executorService)
                .withByteBufferSize(4)
                .map(StandardCharsets.UTF_8::decode)
                ....
```

Reply:
Helidon builder pattern:
- A static factory method `builder` that creates a new builder instance
- Builder has only setters in Helidon style (e.g. no verb)
- There is a clear transition from builder to built instance

So your example:
```java
Multi.builder(is, executorService)
   .bufferSize(1024)
   .build()
   .map...
```

I think it is enough you are breaking the rule for static factory methods, which should always be called `create` and not `from`

Reply:
```java
IoMulti.builder(is)
                .executor(Executors.newSingleThreadExecutor())
                .byteBufferSize(BUFFER_SIZE)
                .build();
```